File: python/depricated.py
import sys


show_deprication_warning = True
deprication_text = """
---
warning: type `{}` is depricated. use `struct` instead.

See https://github.com/jfjlaros/bin-parser/blob/master/README.md#struct for
more information.
---

"""


def deprication_warning(message):
    global show_deprication_warning

    if show_deprication_warning:
        sys.stderr.write(deprication_text.format(message))
    show_deprication_warning = False

File: javascript/depricated.js
'use strict';

var deprication_text = `
---
warning: type \`{}\` is depricated. use \`struct\` instead.

See https://github.com/jfjlaros/bin-parser/blob/master/README.md#basic_types
for more information.
---
`;

function deprication_warning(message) {
  if (this.show_deprication_warning === undefined) {
    console.log(deprication_text.replace('{}', message));
    this.show_deprication_warning = false;
  }
}

module.exports = {
  'deprication_warning': deprication_warning
};
